DOCS-RN-rc2-update-fix

Andriciuc opened the pull request.

Andriciuc committed:
updated rc2

To temp hide indexing until release

Andriciuc committed:
Update mkdocs.yml

Andriciuc committed:
updated with RC2

added RC2 information and added 1 small fix

Andriciuc committed:
Update rc2.md

Andriciuc committed:
updated RC2

With latest notes based on RC2 jira RN.

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -8,33 +8,41 @@
 
-* **Ability to change key provider configuration**
+* **Automatic WAL key rotation**.
 
-    You no longer need to configure a new key provider and set a new principal key if the provider's configuration changed. Now can change the key provider configuration both for the current database and the entire PostgreSQL cluster using [functions](../functions.md#key-provider-management). This enhancement lifts existing limitations and is a native and common way to operate in PostgreSQL.
+    A new automatic WAL key rotation occurs on server start. This ensures each server instance uses a fresh internal key for WAL encryption, improving cryptographic hygiene and reducing the risk of key reuse.
```
The title and the first sentence create an impression that we rotate the principal key. As in fact we don't rotate any key but create a new internal key. Meaning, that everything that happen in WAL before the start remains the same (encryption, keys etc) and all further records in WAL will be encrypted with the new key (obviously, if the WAL encryption is ON)

Andriciuc replied:
Updated, let me know if this is more clear now and thank you very much for the explanation!

dAdAbird replied:
Looks good!

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -8,33 +8,41 @@
+    Previously, encrypted relation keys persisted even after dropping the associated tables, potentially leaving orphaned entries in the map file. This is now corrected, when an encrypted table is dropped, its corresponding key is also removed from the key map.
 
-* **Additional information about principal keys and providers**
+    You can verify the cleanup by running `hexdump -C` on the map file and confirming the related entries are zeroed out.
```
The last sentence was for QAs. I don't think we need this in release notes.

Andriciuc replied:
Removed the sentence, thank you!

Andriciuc committed:
updated Automatic WAL key rotation

Improved wording as new internal key is generated after server start

Andriciuc committed:
Updated **Additional information about principal keys and providers**

Removed last sentence that was for internal use.

Andriciuc committed:
Update rc2.md

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into rn-rc2-update-fix

Andriciuc committed:
Update variables.yml

Andriciuc committed:
Update variables.yml

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-813](https://perconadev.atlassian.net/browse/PG-813) – Enabled support for logical replication in `pg_tde`.
+* [PG-817](https://perconadev.atlassian.net/browse/PG-817) – Added fuzz testing to `pstress` to strengthen validation and resilience.
+* [PG-824](https://perconadev.atlassian.net/browse/PG-824) – Ensured fsync is called on `pg_tde.map`, `pg_tde.dat`, and FS key provider files.
+* [PG-830](https://perconadev.atlassian.net/browse/PG-830) – Implemented full WAL encryption using Vault keyring.
```
PR referenced in PG-830 was merged last July, hence those changes are already got into prev releases.

Andriciuc replied:
Keeping this for transparency in RN

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
-* [PG-1303](https://perconadev.atlassian.net/browse/PG-1303) - Deprecated tde_heap_basic access method.
+* [PG-813](https://perconadev.atlassian.net/browse/PG-813) – Enabled support for logical replication in `pg_tde`.
+* [PG-817](https://perconadev.atlassian.net/browse/PG-817) – Added fuzz testing to `pstress` to strengthen validation and resilience.
+* [PG-824](https://perconadev.atlassian.net/browse/PG-824) – Ensured fsync is called on `pg_tde.map`, `pg_tde.dat`, and FS key provider files.
```
The commit for PG-824 was merged last June. On the other hand, the info probably wasn't included in previous release notes...

Andriciuc replied:
Yep, it is missing from the RC RN so might as well leave it here now I guess.

Andriciuc replied:
Leaving it in

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
-* [PG-1294](https://perconadev.atlassian.net/browse/PG-1294) - Redesigned WAL encryption.
-
-* [PG-1303](https://perconadev.atlassian.net/browse/PG-1303) - Deprecated tde_heap_basic access method.
+* [PG-813](https://perconadev.atlassian.net/browse/PG-813) – Enabled support for logical replication in `pg_tde`.
```
I'd remove PG-813 from this list. The issue was closed as it was related to the legacy functionality. Beside we haven't properly tested Logical replication yet and although it should work I don't think we're committed to any guarantees regarding it for the upcoming release.

Andriciuc replied:
Removed.

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-817](https://perconadev.atlassian.net/browse/PG-817) – Added fuzz testing to `pstress` to strengthen validation and resilience.
+* [PG-824](https://perconadev.atlassian.net/browse/PG-824) – Ensured fsync is called on `pg_tde.map`, `pg_tde.dat`, and FS key provider files.
+* [PG-830](https://perconadev.atlassian.net/browse/PG-830) – Implemented full WAL encryption using Vault keyring.
+* [PG-831](https://perconadev.atlassian.net/browse/PG-831) – Tested WAL recovery and both streaming and logical replication compatibility.
```
The same as  PG-813

Andriciuc replied:
Leaving it in

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-824](https://perconadev.atlassian.net/browse/PG-824) – Ensured fsync is called on `pg_tde.map`, `pg_tde.dat`, and FS key provider files.
+* [PG-830](https://perconadev.atlassian.net/browse/PG-830) – Implemented full WAL encryption using Vault keyring.
+* [PG-831](https://perconadev.atlassian.net/browse/PG-831) – Tested WAL recovery and both streaming and logical replication compatibility.
+* [PG-855](https://perconadev.atlassian.net/browse/PG-855) – Added a contributor guide to help new developers engage with pg_tde.
```
Looks like PG-855 was done before previous release(s)

Andriciuc replied:
Leaving it in

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-830](https://perconadev.atlassian.net/browse/PG-830) – Implemented full WAL encryption using Vault keyring.
+* [PG-831](https://perconadev.atlassian.net/browse/PG-831) – Tested WAL recovery and both streaming and logical replication compatibility.
+* [PG-855](https://perconadev.atlassian.net/browse/PG-855) – Added a contributor guide to help new developers engage with pg_tde.
+* [PG-938](https://perconadev.atlassian.net/browse/PG-938) – Evaluated use of `pg_basebackup` for automated backup validation with pg_tde.
```
Looks like PG-938 was done before previous release

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-855](https://perconadev.atlassian.net/browse/PG-855) – Added a contributor guide to help new developers engage with pg_tde.
+* [PG-938](https://perconadev.atlassian.net/browse/PG-938) – Evaluated use of `pg_basebackup` for automated backup validation with pg_tde.
+* [PG-962](https://perconadev.atlassian.net/browse/PG-962) – Automated test cases to validate data integrity after PostgreSQL restart.
+* [PG-1001](https://perconadev.atlassian.net/browse/PG-1001) – Verified encryption behavior of temporary tables.
```
Looks like PG-1001 was addressed before the last release

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-1099](https://perconadev.atlassian.net/browse/PG-1099) – Developed automation for bare-metal performance benchmarking.
+* [PG-1232](https://perconadev.atlassian.net/browse/PG-1232) – Documented the architectural design behind pg_tde.
+* [PG-1289](https://perconadev.atlassian.net/browse/PG-1289) – Added test cases for verifying compatibility with different PostgreSQL versions.
+* [PG-1295](https://perconadev.atlassian.net/browse/PG-1295) – Introduced TAP test cases to validate WAL encryption across access methods.
```
PG-1295 should be removed from this list, as it was related to the no longer existent functionality

Andriciuc replied:
Removed

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
-* [PG-1361](https://perconadev.atlassian.net/browse/PG-1361) - Fixed pg_tde relocatability.
-
-* [PG-1380](https://perconadev.atlassian.net/browse/PG-1380) - Added support for `pg_tde_is_encrypted()` function on indexes and sequences.
+* [PG-826](https://perconadev.atlassian.net/browse/PG-826) – Documented how to encrypt and decrypt existing tables using pg_tde.
+* [PG-827](https://perconadev.atlassian.net/browse/PG-827) – Fixed CI pipeline tests on the smgr branch.
+* [PG-834](https://perconadev.atlassian.net/browse/PG-834) – Resolved issues with `CREATE ... USING pg_tde` on the smgr branch.
```
Those are either not relevant anymore or implemented before the prev release

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-826](https://perconadev.atlassian.net/browse/PG-826) – Documented how to encrypt and decrypt existing tables using pg_tde.
+* [PG-827](https://perconadev.atlassian.net/browse/PG-827) – Fixed CI pipeline tests on the smgr branch.
+* [PG-834](https://perconadev.atlassian.net/browse/PG-834) – Resolved issues with `CREATE ... USING pg_tde` on the smgr branch.
+* [PG-838](https://perconadev.atlassian.net/browse/PG-838) – Added documentation for setting up streaming replication with pg_tde.
```
Dup of 802

Andriciuc replied:
Removed

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-827](https://perconadev.atlassian.net/browse/PG-827) – Fixed CI pipeline tests on the smgr branch.
+* [PG-834](https://perconadev.atlassian.net/browse/PG-834) – Resolved issues with `CREATE ... USING pg_tde` on the smgr branch.
+* [PG-838](https://perconadev.atlassian.net/browse/PG-838) – Added documentation for setting up streaming replication with pg_tde.
+* [PG-1294](https://perconadev.atlassian.net/browse/PG-1294) – Improved WAL encryption design and performance.
```
I'd base it on the commit description: https://github.com/percona/postgres/commit/d352e5af41f2f5ce449434d33a2baa1e4dbe6b44

Andriciuc replied:
Updated the text descriptor to: 

Changed the approach to WAL encryption. Instead of
encrypting each WAL page and keeping its header unencrypted with a special
encrypted flag, this change moves encrypted/unencrypted bookkeeping to
WAL internal keys. Now every [WAL] internal key has an additional field
`start_lsn`.

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-838](https://perconadev.atlassian.net/browse/PG-838) – Added documentation for setting up streaming replication with pg_tde.
+* [PG-1294](https://perconadev.atlassian.net/browse/PG-1294) – Improved WAL encryption design and performance.
+* [PG-1304](https://perconadev.atlassian.net/browse/PG-1304) – Removed deprecated `tde_heap_basic` access method before GA.
+* [PG-1307](https://perconadev.atlassian.net/browse/PG-1307) – Assessed overall development effort required for WAL encryption.
```
PG-1307 doesn't sound like it's should be in the release notes

Andriciuc replied:
True, removed.

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-1392](https://perconadev.atlassian.net/browse/PG-1392) – Investigated encryption coverage gaps in freespace and visibility maps.
+* [PG-1419](https://perconadev.atlassian.net/browse/PG-1419) – Changed provider validation to occur during provider addition instead of key setup.
+* [PG-1427](https://perconadev.atlassian.net/browse/PG-1427) – Tested and fixed KMIP implementation for Thales support.
+* [PG-1437](https://perconadev.atlassian.net/browse/PG-1437) – Cleaned up and resolved TODO items in the codebase.
```
PG-1437 sounds like an internal tasks that doesn't have to be in the release notes. It makes sense to include some of PRs implemented for this ticket. But current description is too vague for the release notes.

Andriciuc replied:
Removed, thank you!

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-1427](https://perconadev.atlassian.net/browse/PG-1427) – Tested and fixed KMIP implementation for Thales support.
+* [PG-1437](https://perconadev.atlassian.net/browse/PG-1437) – Cleaned up and resolved TODO items in the codebase.
+* [PG-1440](https://perconadev.atlassian.net/browse/PG-1440) – Restricted key provider configuration to superusers for improved security.
+* [PG-1441](https://perconadev.atlassian.net/browse/PG-1441) – Prevented replication of table-level encryption keys.
```
It's not like we prevented something but rather changed the approach of how and what we replicate

Andriciuc replied:
Reworded to: Changed the approach for replicating table-level encryption keys.

dAdAbird commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -1,121 +1,129 @@
+* [PG-1446](https://perconadev.atlassian.net/browse/PG-1446) – Improved locking mechanisms in pg_tde.
+* [PG-1447](https://perconadev.atlassian.net/browse/PG-1447) – Verified compatibility of encryption with template databases.
+* [PG-1457](https://perconadev.atlassian.net/browse/PG-1457) – Renamed key management functions for clarity and consistency.
+* [PG-1467](https://perconadev.atlassian.net/browse/PG-1467) – Added Clang-based CI integration on GitHub.
```
Sounds like an internal task for me

Andriciuc replied:
Removed the four PG's.

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into rn-rc2-update-fix

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into rn-rc2-update-fix

Andriciuc committed:
Merge branch 'rn-rc2-update-fix' of https://github.com/percona/postgres into rn-rc2-update-fix

Andriciuc committed:
Update rc2.md

Updated based on Andrew's feedback

Andriciuc committed:
Merge branch 'TDE_REL_17_STABLE' into rn-rc2-update-fix

nastena1606 commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -82,20 +80,14 @@ Adjust the limits with caution since it affects other processes running in your
-* [PG-838](https://perconadev.atlassian.net/browse/PG-838) – Added documentation for setting up streaming replication with pg_tde.
-* [PG-1294](https://perconadev.atlassian.net/browse/PG-1294) – Improved WAL encryption design and performance.
+* [PG-1294](https://perconadev.atlassian.net/browse/PG-1294) – Changed the approach to WAL encryption. Instead of
+encrypting each WAL page and keeping its header unencrypted with a special encrypted flag, this change moves encrypted/unencrypted bookkeeping to WAL internal keys. Now every [WAL] internal key has an additional field `start_lsn`.
```
Why [WAL] in brackets? Should it be a link?

nastena1606 replied:
This item was in RC release notes. Is it not finalized? Why to repeat it here?

Andriciuc replied:
True! I removed this, also cleaned the duplicates and fixed the banner for RC 2

Andriciuc committed:
updates to banner and duplicates

Removed duplicate tickets from RC 1 to 2, updated banner and fixed an issue with missing () in ToC
